src/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(QUICK_CUDA_CUDA_SOURCE gpu.cu gpu_type.cu gpu_get2e.cu)
+     
+set(QUICK_CUDA_FORTRAN_SOURCE gpu_write_info.f90)
+
+set_property(SOURCE ${QUICK_CUDA_FORTRAN_SOURCE} PROPERTY COMPILE_OPTIONS ${OPT_FFLAGS})
+
+#sadly FindCUDA doesn't support OBJECT libraries, so we have to use a static cuda_add_library
+cuda_add_library(quick_cuda STATIC ${QUICK_CUDA_FORTRAN_SOURCE} ${QUICK_CUDA_CUDA_SOURCE} 
+	OPTIONS ${CUDA_DEVICE_CODE_FLAGS} -DCUDA) 
+target_compile_options(quick_cuda PRIVATE -DCUDA)
+target_link_libraries(quick_cuda xc_gpu xc_cuda)
+make_pic_if_needed(quick_cuda)
+
+if(MPI)
+	cuda_add_library(quick_cuda_mpi STATIC ${QUICK_CUDA_FORTRAN_SOURCE} ${QUICK_CUDA_CUDA_SOURCE} 
+	OPTIONS ${CUDA_DEVICE_CODE_FLAGS} -DMPIV -DCUDA_MPIV) 
+	target_compile_options(quick_cuda_mpi PRIVATE -DMPIV -DCUDA_MPIV)
```

Comment:
Shouldn't here be `target_compile_definitions(quick_cuda_mpi PRIVATE -DMPIV -DCUDA_MPIV)`?

Reply:
Yeah that's more correct I suppose, except you don't use the -D flag there so it becomes
```cmake
target_compile_definitions(quick_cuda_mpi PRIVATE MPIV CUDA_MPIV)
```
The only thing is, FindCUDA is weird about compile options -- it doesn't apply them to .cu sources.  The only options that are actually applied are the ones in the OPTIONS argument above.